src/main/java/com/xceptance/neodymium/util/SelenideAddons.java

```diff
@@ -410,8 +411,21 @@ public static void wrapAssertionError(final Runnable runnable)
      */
     public static void dragAndDrop(SelenideElement elementToMove, int horizontalMovement, int verticalMovement)
     {
-        // perform drag and drop via the standard Selenium way
-        new Actions(Neodymium.getDriver()).dragAndDropBy(elementToMove.getWrappedElement(), horizontalMovement, verticalMovement).build().perform();
+        try
+        {
+            // perform drag and drop via the standard Selenium way
+            new Actions(Neodymium.getDriver()).dragAndDropBy(elementToMove.getWrappedElement(), horizontalMovement, verticalMovement).build().perform();
+        }
+        catch (MoveTargetOutOfBoundsException targetOutOfBound)
+        {
+            String parameterMessage = horizontalMovement != 0 ? (verticalMovement != 0 ? "'horizontalMovement' and 'verticalMovement'" : "'horizontalMovement'")
+                                                              : "'verticalMovement'";
+
+            throw UIAssertionError.wrap(WebDriverRunner.driver(),
```

Comment:
Please use a `WebDriverException` instead of an `AssertionError`

Reply:
Done. The `UIAssertionError.wrap` returns `Throwable` instead of `Error` if the input parameter is `WebDriverException`. As we want to avoid making users put this method call in the try-catch clause, I wrapped the returned `Throwable` object into `RuntimeException`. I'm not sure, if it's the correct approach, @occupant23, please advise.

Btw, why do we need `WebDriverException` and why should we wrap it into `UIAssertionError`?

Reply:
@oomelianchuk 
The `MoveTargetOutOfBoundsException` extends `WebDriverException`.
We don't need an extra wrapping as `RuntimeException` we want to implement a similar behavior as Selenide in this utility class. Btw `Error` is the one path that extends from `Throwable` while `RuntimeException` extends `Exception` which is the other path that derives from `Throwable`.

Reply:
Discussed and done